agent.go

```diff
@@ -463,52 +463,64 @@ func (ep *endpoint) deleteDriverInfoFromCluster() error {
 }
 
 func (ep *endpoint) addServiceInfoToCluster() error {
+	if ep.isAnonymous() && len(ep.myAliases) == 0 {
+		return nil
+	}
+
 	n := ep.getNetwork()
 	if !n.isClusterEligible() {
 		return nil
 	}
 
 	c := n.getController()
 	agent := c.getAgent()
-	if !ep.isAnonymous() && ep.Iface().Address() != nil {
```

Comment:
`ep.isAnonymous() check has been moved up, which is needed. But why is the nil check for `ep.Iface().Address()` removed ?